Std/Tactic/Simps/Basic.lean

```diff
@@ -0,0 +1,1157 @@
+  See `initialize_simps_projections`. -/
+inductive ProjectionRule where
+  /-- A renaming rule `before→after` or
+    Each name comes with the syntax used to write the rule,
+    which is used to declare hover information. -/
+  | rename (oldName : Name) (oldStx : Syntax) (newName : Name) (newStx : Syntax) :
+      ProjectionRule
+  /-- An adding rule `+fieldName` -/
+  | add : Name → Syntax → ProjectionRule
+  /-- A hiding rule `-fieldName` -/
+  | erase : Name → Syntax → ProjectionRule
+  /-- A prefix rule `prefix fieldName` -/
+  | prefix : Name → Syntax → ProjectionRule
+
+instance : ToMessageData ProjectionRule where toMessageData
+  | .rename x₁ x₂ x₃ x₄ => .group <| .nest 1 <|
+    "rename ⟨" ++ .joinSep [toMessageData x₁, toMessageData x₂, toMessageData x₃, toMessageData x₄]
+      ("," ++ Format.line) ++ "⟩"
+  | .add x₁ x₂ => .group <| .nest 1 <|
+    "+⟨" ++ .joinSep [toMessageData x₁, toMessageData x₂] ("," ++ Format.line) ++ "⟩"
+  | .erase x₁ x₂ => .group <| .nest 1 <|
+    "-⟨" ++ .joinSep [toMessageData x₁, toMessageData x₂] ("," ++ Format.line) ++ "⟩"
+  | .prefix x₁ x₂ => .group <| .nest 1 <|
+    "prefix ⟨" ++ .joinSep [toMessageData x₁, toMessageData x₂] ("," ++ Format.line) ++ "⟩"
+
+/-- Returns the projection information of a structure. -/
+def projectionsInfo (l : List ProjectionData) (pref : String) (str : Name) : MessageData :=
+  let ⟨defaults, nondefaults⟩ := l.partition (·.isDefault)
+  let toPrint : List MessageData :=
+    defaults.map fun s =>
+      let prefixStr := if s.isPrefix then "(prefix) " else ""
+      m!"Projection {prefixStr}{s.name}: {s.expr}"
+  let print2 : MessageData :=
+    String.join <| (nondefaults.map fun nm : ProjectionData => toString nm.1).intersperse ", "
+  let toPrint :=
+    toPrint ++
+      if nondefaults.isEmpty then [] else
+      [("No lemmas are generated for the projections: " : MessageData) ++ print2 ++ "."]
+  let toPrint := MessageData.joinSep toPrint ("\n" : MessageData)
+  m!"{pref} {str}:\n{toPrint}"
+
+/-- Find the indices of the projections that need to be applied to elaborate `$e.$projName`.
+Example: If `e : α ≃+ β` and ``projName = `invFun`` then this returns `[0, 1]`, because the first
+projection of `MulEquiv` is `toEquiv` and the second projection of `Equiv` is `invFun`. -/
+def findProjectionIndices (strName projName : Name) : MetaM (List Nat) := do
+  let env ← getEnv
+  let .some baseStr := findField? env strName projName |
+    throwError "{strName} has no field {projName} in parent structure"
+  let .some fullProjName := getProjFnForField? env baseStr projName |
+    throwError "no such field {projName}"
+  let .some pathToField := getPathToBaseStructure? env baseStr strName |
+    throwError "no such field {projName}"
+  let allProjs := pathToField ++ [fullProjName]
+  return allProjs.map (env.getProjectionFnInfo? · |>.get!.i)
+
+/-- Auxiliary function of `getCompositeOfProjections`. -/
```

Comment:
Could more detail be added either here to body of code?

Reply:
@fpvandoorn, could I perhaps interest you in this? :-)

Reply:
Oh, I don't have push access to this repo. Can I get access, or shall I make a PR into this PR?

Reply:
@fpvandoorn, for now, would you mind a PR to the PR? Really I should have made this PR from a fork so I could have given you access myself.